### PR TITLE
[[ Tests ]] Standalone test runner

### PR DESCRIPTION
**Pull request by montegoulding**

### CODE REVIEW

**Comment by montegoulding** [comment]
@runrevmark @livecodeali This should be good to go now. I'm leaving as WIP until dependant patches have been merged and I can drop those commits from this patch.

For reference the dependant PRs are:
- #5560
- #5572

Some other patches remain in this PR until they are merged up from `develop-8.1`

**Comment by montegoulding** [comment]
If this is merged as is we should open the following enhancement requests:

 - A method to get data back for Android & iOS (sockets?)
 - Run Android (allow specifying emulator or connected device)
 - Build iOS allow specifying provisioning profile
 - Run iOS (allow specifying simulator or use mergDeploy for connected device)
 - Revisit mergDeploy automatic run after install
 - Build & Run in server
 - Add a build target for development that builds a target platform & host platform then builds and runs the target platform standalone

**Comment by montegoulding** [comment]
This patch required the IDE patch https://github.com/livecode/livecode-ide/pull/1635

File: tests/lcs/core/engine/target.livecodescript [hunk 1]
```diff
@@ -58,3 +58,12 @@ end TestDeleteTargetInOwner
...
 end __TestDeleteTarget
+
+on TestTeardown
+   if there is a button "test" then
+      delete button "test"
+   end if
+   if there is a button "script" then
+      delete button "script"
+   end if
+end TestTeardown
```
Comment (livecodeali) [inline tests/lcs/core/engine/target.livecodescript]: Is this needed anymore?

File: tests/lcs/core/interface/interfaceui.livecodescript [hunk 2]
```diff
@@ -294,5 +348,12 @@ push current card
...
 TestAssert "test", field 1 is the long id of the current card
+delete stack "Test"
 
 end TestInterface85
+
+on TestTeardown
+   if there is a stack "Test" then
+      delete stack "Test"
+   end if
+end TestTeardown
```
Comment (livecodeali) [inline tests/lcs/core/interface/interfaceui.livecodescript]: None of these `delete stack`s are needed anymore are they?

File: tests/lcs/core/interface/interfaceui.livecodescript [hunk 3]
```diff
@@ -37,16 +37,22 @@ get the formattedRect of char 1 of field "tf"
 local tPoint
 put (item 1 of it + item 3 of it) div 2 & "," & (item 2 of it + item 4 of it) div 2 into tPoint
 
-click at tPoint
-TestAssert "test", the clickChar is "a"
+click button 3 at tPoint
+if TestIsInStandalone() then
+   TestAssertBroken "the clickChar is a", the clickChar is "a", "not sure"
```
Comment (livecodeali) [inline tests/lcs/core/interface/interfaceui.livecodescript:42]: What's the deal with this "not sure"? I think it's a bug if these aren't working in a standalone isn't it? In which case the reason should be the bug number. Or are you not sure if it is a bug or not?

File: tests/lcs/core/interface/interfaceui.livecodescript (hunk 2 above)
Comment (montegoulding) [reply tests/lcs/core/interface/interfaceui.livecodescript]: no the probably aren't

File: tests/lcs/core/interface/interfaceui.livecodescript (hunk 3 above)
Comment (montegoulding) [reply tests/lcs/core/interface/interfaceui.livecodescript:42]: lol... yeah I really wasn't sure what was going on there. I will try testing it again to see if I can work it out.

**Comment by livecodeali** [comment]
@livecode-vulcan review ok 99941be

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok 99941beb3985fb4a7afade2ceb7f188ffa96008c

**Comment by livecode-vulcan** [comment]
😞 test failure 99941beb3985fb4a7afade2ceb7f188ffa96008c

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/535)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/550)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/524)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/467)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/470)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/530)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/538)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/533)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/550)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/533)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/543)
* try-community-x86-win32: [failure](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/565)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/529)

**Comment by livecodeali** [comment]
@livecode-vulcan retry 99941be

**Comment by livecode-vulcan** [comment]
😞 test failure 99941beb3985fb4a7afade2ceb7f188ffa96008c

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/537)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/552)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/526)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/469)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/472)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/532)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/540)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/535)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/552)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/535)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/545)
* try-community-x86-win32: [failure](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/567)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/531)

**Comment by livecodeali** [comment]
@livecode-vulcan retry 99941be

**Comment by livecode-vulcan** [comment]
😎 test success 99941beb3985fb4a7afade2ceb7f188ffa96008c

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/539)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/554)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/528)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/471)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/474)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/534)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/542)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/537)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/554)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/537)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/547)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/569)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/533)

**Pull request merged**